### PR TITLE
[addon] Move add-on library parts to independent folders

### PR DESCRIPTION
**Pull request by AlwinEsch**
Again a request for binary add-on lib rework.

This does only move the current library parts on Kodi to new folders and change the namespace in kodi itself to new style (`V1::KodiAPI::...`). 

On Add-ons is nothing to be changed and is still compatibel to before compiled add-ons!

This request is important for me to observe better changes on libraries before API level 2 comes in and to prevent miss of parts if it becomes updated only on my rework side!

Please check the folder structure is acceptable.

### CODE REVIEW

File: lib/addons/library.kodi.audioengine/libKODI_audioengine.cpp [hunk 1]
```diff
@@ -224,8 +225,4 @@ void CAddonAEStream::SetResampleRatio(double Ratio)
   ((CB_AudioEngineLib*)m_Callbacks)->AEStream_SetResampleRatio(((AddonCB*)m_AddonHandle)->addonData, m_StreamHandle, Ratio);
 }
 
-void CAddonAEStream::Discontinuity()
```
Comment (AlwinEsch) [inline lib/addons/library.kodi.audioengine/libKODI_audioengine.cpp]: Just a note, this function was not removed on last update here and related part on Kodi was already missing.

Comment (AlwinEsch) [inline lib/addons/library.kodi.audioengine/libKODI_audioengine.cpp]: @AchimTuran unused function correct?

**Comment by AlwinEsch** [comment]
jenkins build this please

**Comment by AlwinEsch** [comment]
Can a windows guy help me about this error? Is this from C++11 not supported?

- Code:
```cpp
private:
  static constexpr const int   m_apiLevel = 1;
  static constexpr const char* m_version  = "0.0.1";
```
- Failure:
```
     7>c:\jenkins\slave\workspace\WIN-32\xbmc\addons/binary/interfaces/api1/Addon/AddonCallbacksAddon.h(174): error C2143: syntax error : missing ';' before 'const' [c:\jenkins\slave\workspace\WIN-32\project\VS2010Express\XBMC.vcxproj]
     7>c:\jenkins\slave\workspace\WIN-32\xbmc\addons/binary/interfaces/api1/Addon/AddonCallbacksAddon.h(174): error C4430: missing type specifier - int assumed. Note: C++ does not support default-int [c:\jenkins\slave\workspace\WIN-32\project\VS2010Express\XBMC.vcxproj]
     7>c:\jenkins\slave\workspace\WIN-32\xbmc\addons/binary/interfaces/api1/Addon/AddonCallbacksAddon.h(175): error C2143: syntax error : missing ';' before 'const' [c:\jenkins\slave\workspace\WIN-32\project\VS2010Express\XBMC.vcxproj]
     7>c:\jenkins\slave\workspace\WIN-32\xbmc\addons/binary/interfaces/api1/Addon/AddonCallbacksAddon.h(175): error C4430: missing type specifier - int assumed. Note: C++ does not support default-int [c:\jenkins\slave\workspace\WIN-32\project\VS2010Express\XBMC.vcxproj]
     7>c:\jenkins\slave\workspace\WIN-32\xbmc\addons/binary/interfaces/api1/Addon/AddonCallbacksAddon.h(175): error C2086: 'int V1::KodiAPI::AddOn::CAddonCallbacksAddon::constexpr' : redefinition [c:\jenkins\slave\workspace\WIN-32\project\VS2010Express\XBMC.vcxproj]
```

There is also a failure on MacOS X but maybe I find it:
```
/usr/bin/clang++ -fPIC -O2 -g -D_DEBUG -Wall -no-cpp-precomp -arch x86_64 -mmacosx-version-min=10.7 -std=c++11 -stdlib=libc++ -g -O2 -std=gnu++11 -isysroot /Applications/Xcode6.1.1.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX10.10.sdk -isystem /Users/Shared/jenkins/workspace/OSX-64/tools/depends/xbmc-depends/macosx10.10_x86_64-target/include -no-cpp-precomp -arch x86_64 -mmacosx-version-min=10.7 -std=c++11 -stdlib=libc++ -g -O2 -std=gnu++11 -isysroot /Applications/Xcode6.1.1.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX10.10.sdk -isystem /Users/Shared/jenkins/workspace/OSX-64/tools/depends/xbmc-depends/macosx10.10_x86_64-target/include -O3  -L/Users/Shared/jenkins/workspace/OSX-64/tools/depends/xbmc-depends/macosx10.10_x86_64-target/lib -Wl,-search_paths_first -arch x86_64 -mmacosx-version-min=10.7 -isysroot /Applications/Xcode6.1.1.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX10.10.sdk -stdlib=libc++    -dynamiclib -o ../../../addons/library.kodi.guilib/libKODI_guilib-x86-osx.dylib libKODI_guilib.o     
Undefined symbols for architecture x86_64:
  "CGraphicContext::CGraphicContext()", referenced from:
      __GLOBAL__I_a in libKODI_guilib.o
ld: symbol(s) not found for architecture x86_64
clang: error: linker command failed with exit code 1 (use -v to see invocation)
make[1]: *** [../../../addons/library.kodi.guilib/libKODI_guilib-x86-osx.dylib] Error 1
make: *** [libaddon] Error 2
make: *** Waiting for unfinished jobs....
```

**Comment by Montellese** [comment]
VS2013's compiler doesn't support constexpr.

**Comment by AlwinEsch** [comment]
Thanks @Montellese 
have only VS2015 here :)

**Comment by AlwinEsch** [comment]
jenkins build this please

**Comment by AlwinEsch** [comment]
Boaah, _stat64 on add-on is like Roulette in Casino :(

jenkins build this please

**Comment by AlwinEsch** [comment]
jenkins build this please

**Comment by AlwinEsch** [comment]
jenkins build this please

**Comment by AlwinEsch** [comment]
jenkins build this please again

**Comment by AlwinEsch** [comment]
@ksooo, @FernetMenta acceptable to merge in now (if jenkins say OK)???
No changes on addon type systems itself, only the libraries to kodi itself.

**Comment by MartijnKaijser** [comment]
Fix merge conflicts first

**Comment by AlwinEsch** [comment]
Thanks a lot @MartijnKaijser 
The windows build is fixed, was come in on my Mac OS X update before :(

**Comment by AlwinEsch** [comment]
jenkins build this please again

**Comment by FernetMenta** [comment]
@AlwinEsch did you use git mv? some files seem to have lost history.

**Comment by AlwinEsch** [comment]
Thanks @FernetMenta yes, put possible it was lost during the last updates where git resets was needed.
Has not looked on end over the history. I do re commit of them to fix it.

**Comment by AlwinEsch** [comment]
Is separated in two parts. With one the history was not OK.

jenkins build this please

**Pull request merged**